pkg/process/net/common.go

```diff
@@ -126,34 +129,64 @@ func (r *RemoteSysProbeUtil) GetProcStats(pids []int32) (*model.ProcStatsWithPer
 	return results, nil
 }
 
-// GetConnections returns a set of active network connections, retrieved from the system probe service
+// GetConnections returns a set of active network connections, retrieved from the system probe service, supports the
+// grpc server as well as the http server.
 func (r *RemoteSysProbeUtil) GetConnections(clientID string) (*model.Connections, error) {
-	req, err := http.NewRequest("GET", fmt.Sprintf("%s?client_id=%s", connectionsURL, clientID), nil)
-	if err != nil {
-		return nil, err
-	}
-
-	req.Header.Set("Accept", contentTypeProtobuf)
-	resp, err := r.httpClient.Do(req)
-	if err != nil {
-		return nil, err
-	}
-
-	defer resp.Body.Close()
-
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("conn request failed: Probe Path %s, url: %s, status code: %d", r.path, connectionsURL, resp.StatusCode)
-	}
-
-	body, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return nil, err
-	}
-
-	contentType := resp.Header.Get("Content-type")
-	conns, err := netEncoding.GetUnmarshaler(contentType).Unmarshal(body)
-	if err != nil {
-		return nil, err
+	var conns *model.Connections
+	useGRPCServer := dd_config.SystemProbe.GetBool("service_monitoring_config.use_grpc")
+	if useGRPCServer {
+		conn, err := grpc.Dial("unix:///tmp/my_grpc.sock", grpc.WithInsecure())
+		if err != nil {
+			log.Errorf("Failed to connect: %v", err)
+		}
+
+		client := test2.NewSystemProbeClient(conn)
+
+		response, err := client.GetConnections(context.Background(), &test2.GetConnectionsRequest{ClientID: clientID})
```

Comment:
It will be best to have a timed context here, otherwise we can be stuck for a long time

Reply:
create a 15 sec time out